presto-hive/src/test/java/com/facebook/presto/hive/orc/TestOrcBatchPageSourceFactory.java

```diff
@@ -0,0 +1,103 @@
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+public class TestOrcBatchPageSourceFactory
+{
+    @Test
+    public void testCreateOrcPageSourceException() throws Exception
+    {
+        Random random = new Random();
+        List<TestOrcBatchPageSourceMemoryTracking.TestColumn> testColumns = ImmutableList.<TestOrcBatchPageSourceMemoryTracking.TestColumn>builder()
+                .add(new TestOrcBatchPageSourceMemoryTracking.TestColumn("p_empty_string", javaStringObjectInspector, () -> "", true))
+                .add(new TestOrcBatchPageSourceMemoryTracking.TestColumn("p_string", javaStringObjectInspector, () -> Long.toHexString(random.nextLong()), false))
+                .build();
+        File tempFile = File.createTempFile("presto_test_orc_batch_page_source_factory", "orc");
+        tempFile.delete();
+        ImmutableList.Builder<HiveColumnHandle> columnsBuilder = ImmutableList.builder();
+        Configuration configuration = new JobConf(new Configuration(false));
+        FileSplit fileSplit = TestOrcBatchPageSourceMemoryTracking.createTestFile(tempFile.getAbsolutePath(),
+                new OrcOutputFormat(), new OrcSerde(), null, testColumns, 1, 1);
+        HiveFileSplit hiveFileSplit = new HiveFileSplit(
+                fileSplit.getPath().toString(),
+                fileSplit.getStart(),
+                fileSplit.getLength(),
+                fileSplit.getLength(),
+                Instant.now().toEpochMilli(),
+                Optional.empty(),
+                ImmutableMap.of(),
+                0);
+        OrcReaderOptions orcReaderOptions = null; //need to be set to null to trigger PrestoException
+        Optional<EncryptionInformation> encryptionInformation =
+                Optional.of(EncryptionInformation.fromEncryptionMetadata(DwrfEncryptionMetadata.forPerField(
+                ImmutableMap.of("field1", "test1".getBytes()),
+                ImmutableMap.of(),
+                "test_algo",
+                "test_provider")));
+        try (ConnectorPageSource connectorPageSource = OrcBatchPageSourceFactory.createOrcPageSource(ORC, HDFS_ENVIRONMENT,
+                configuration, hiveFileSplit, columnsBuilder.build(), true,
+                TupleDomain.all(), DateTimeZone.getDefault(), FUNCTION_AND_TYPE_MANAGER, true,
+                new FileFormatDataSourceStats(), 100, new StorageOrcFileTailSource(),
+                StripeMetadataSourceFactory.of(new StorageStripeMetadataSource()), DEFAULT_HIVE_FILE_CONTEXT, orcReaderOptions,
+                encryptionInformation, HiveDwrfEncryptionProvider.NO_ENCRYPTION.toDwrfEncryptionProvider(), SESSION, Optional.empty())) {
+            fail("createOrcPageSource call should have resulted in an PrestoException.");
+        }
+        catch (PrestoException pe) {
+            assertTrue(pe.getMessage().contains("orcReaderOptions is null"), "Incorrect PrestoException was thrown.");
+        }
+        catch (IOException e) {
```

Comment:
simpler to let the IOexception be thrown which would also fail the test

Reply:
My thinking was that since IOException was the other exception that was explicitly caught and wrapped in a PrestoException I wanted a fail statement if an IOException was thrown eg: "escaped". Now after writing this comment I guess I should have been catching all exception types, so you are right should just let it be thrown for simplicity sake.